algoliasearch/src/main/java/com/algolia/search/saas/BaseAPIClient.java

```diff
@@ -398,8 +421,20 @@ private JSONObject _getAnswerJSONObject(InputStream istream) throws IOException,
      * @throws AlgoliaException if the request data is not valid json
      */
     private synchronized JSONObject _request(Method m, String url, String json, List<String> hostsArray, int connectTimeout, int readTimeout) throws AlgoliaException {
+        String cacheKey = null;
+        if (isCacheEnabled) {
+            cacheKey = String.format("%s:%s(%s)", m, url, json);
+            final JSONObject value = cache.get(cacheKey);
+            if (value != null) {
+                return value;
+            }
+        }
         try {
-            return _getJSONObject(_requestRaw(m, url, json, hostsArray, connectTimeout, readTimeout));
+            final JSONObject result = _getJSONObject(_requestRaw(m, url, json, hostsArray, connectTimeout, readTimeout));
+            if (isCacheEnabled) {
+                cache.put(cacheKey, result);
```

Comment:
I think I would rather store the `string` (raw) version of the JSON answer and regenerate the `JSONObject` every time to ensure we're always returning a new object. We had a bunch of issues not doing it in JavaScript at the beginning and as soon as the JSON object is modified (can happen at display time, for convenience purpose) it's a mess to debug.